File: Services/TaskService/TaskService.Application/UseCases/ResourceUseCases/AddResource/AddResourceCommand.cs
using MediatR;
using Microsoft.AspNetCore.Http;
using TaskService.Domain.Enums;

namespace TaskService.Application.UseCases
{
    public record AddResourceCommand(
        string? username,
        int? companyId,
        int? taskInfoId,
        string? resourceData,
        IFormFile? imageFile,
        ResourceType? type
    ) : IRequest<Unit>;
}

File: FrontendApp/src/app/models/commands/taskservice/add-resource.command.ts
import { ResourceType } from "../../../enums/resource-type.enum";

export interface AddResourceCommand {
  username: string | null;
  companyId: number | null;
  taskInfoId: number | null;
  resourceData: string | null;
  imageFile: File | null;
  type: ResourceType | null;
}
